### PR TITLE
Clambc compiler 1.4.0

### PR DESCRIPTION
**Pull request by micahsnyder**
docker image update for https://github.com/Cisco-Talos/clamav-bytecode-compiler/pull/48

### CODE REVIEW

**Comment by micahsnyder** [comment]
Note: I used ubuntu:23.04 for easy access to clang-16, rather than figuring out the minimal way to install it using LLVM's packages. 
I'm totally up for changing that and picking an LTS image like 22.04.

File: clamav-bytecode-compiler/1.3/ubuntu/Jenkinsfile
Comment (rsundriyal) [inline clamav-bytecode-compiler/1.3/ubuntu/Jenkinsfile:1]: This file needs changes. 
Refer [clambc-1.3.0](https://github.com/rsundriyal/clamav-docker/tree/clambc-1.3.0)

**Comment by rsundriyal** [comment]
If we change the version of docker ubuntu base from 20.04 to (22.04,23.04, we might need to change it in Jenkinsfile for unstable pipeline too. Here -> [File](https://github.com/Cisco-Talos/clamav-docker/blob/f29a7b8193ce497d075368910b756efecbf4047b/clamav-bytecode-compiler/unstable/ubuntu/Jenkinsfile#L53)

**Comment by micahsnyder** [comment]
@rsundriyal Fixed

File: clamav-bytecode-compiler/1.4/ubuntu/Dockerfile [hunk 1]
```diff
@@ -19,15 +19,13 @@ RUN apt-get update -y && \
         binutils \
         git \
         python3 \
-        python3-distutils \
-        python3-pip \
```
Comment (ragusaa) [inline clamav-bytecode-compiler/1.4/ubuntu/Dockerfile:23]: I don't understand why, but removing the python3-pip gives me warnings that pytest is not installed.

**Review by ragusaa — APPROVED** [review]

**Pull request merged**